src/config/backends/extra-uEnv.ts

```diff
@@ -0,0 +1,261 @@
+		);
+		// Check if we found a corresponding option for this entry
+		if (typeof optionKey !== 'string') {
+			log.warn(`Could not parse unsupported option: ${optionArray[0]}`);
+			return {};
+		}
+		return { [optionKey]: VALUE };
+	}
+
+	private static parseOptionCollection(
+		collectionString: string,
+	): ConfigOptions {
+		return (
+			collectionString
+				// Split collection into individual options
+				.split(' ')
+				// Reduce list of option strings into ConfigOptions object
+				.reduce((options: ConfigOptions, option: string) => {
+					// Match optionValues to key=value regex
+					const optionValues = option.match(OPTION_REGEX);
+					// Check if option is only a key
+					if (optionValues === null) {
+						if (option !== '') {
+							return { [option]: '', ...options };
+						} else {
+							log.warn(`Unable to set empty value option: ${option}`);
+							return options;
+						}
+					}
+					const [, KEY, VALUE] = optionValues;
+					// Merge new option with existing options
+					return { [KEY]: VALUE, ...options };
+				}, {})
+		);
+	}
+
+	private static async readBootConfigPath(): Promise<string> {
+		try {
+			return await fs.readFile(ExtraUEnvConfigBackend.bootConfigPath, 'utf-8');
+		} catch {
+			// In the rare case where the user might have deleted extra_uEnv conf file between linux boot and supervisor boot
+			// We do not have any backup to fallback too; warn the user of a possible brick
+			log.error(
+				`Unable to read extra_uEnv file at: ${ExtraUEnvConfigBackend.bootConfigPath}`,
+			);
+			throw new ExtraUEnvError(
+				'Could not find extra_uEnv file. Device is possibly bricked',
+			);
+		}
+	}
+
+	private static configToString(configs: ConfigOptions): string {
+		// Get Map of ConfigOptions object
+		const configMap = ExtraUEnvConfigBackend.configToMap(configs);
+		// Iterator over configMap and concat to configString
+		let configString = '';
+		for (const [key, value] of configMap) {
+			if (!value) {
+				log.warn(`Could not find value for ${key} in configMap.`);
+				continue;
+			}
+			// Append new config
+			configString += `${key}=${value}\n`;
+		}
+		return configString;
+	}
+
+	private static configToMap(configs: ConfigOptions): Map<string, string> {
+		// Reduce ConfigOptions into a Map that joins collections
+		return Object.entries(configs).reduce(
+			(configMap: Map<string, string>, [configKey, configValue]) => {
+				const {
+					key: ENTRY_KEY,
+					collection: ENTRY_IS_COLLECTION,
+				} = ExtraUEnvConfigBackend.supportedConfigs[configKey];
+				// Check if we have to build the value for the entry
+				if (ENTRY_IS_COLLECTION) {
+					return configMap.set(
+						ENTRY_KEY,
+						// Append existing values if already set
+						(configMap.has(ENTRY_KEY) ? `${configMap.get(ENTRY_KEY)} ` : '') +
+							// Add new config's key
+							configKey +
+							// Append the config's value if it has one
+							(configValue !== '' ? `=${configValue}` : ''),
```

Comment:
There's a little bit too much going on here for my liking, I think even just another variable with a descriptive name would clear this up a little

Reply:
Refactored it into another function.